src/realm/object-store/sync/sync_session.cpp

```diff
@@ -393,20 +397,34 @@ std::function<void(util::Optional<app::AppError>)> SyncSession::handle_refresh(s
     };
 }
 
-SyncSession::SyncSession(SyncClient& client, std::string realm_path, SyncConfig config, bool force_client_resync)
+SyncSession::SyncSession(SyncClient& client, std::string realm_path, SyncConfig config, bool force_client_resync,
+                         SyncManager* sync_manager)
     : m_state(&State::inactive)
     , m_config(std::move(config))
     , m_force_client_resync(force_client_resync)
     , m_realm_path(std::move(realm_path))
     , m_client(client)
+    , m_sync_manager(sync_manager)
+{
+}
+
+std::shared_ptr<SyncManager> SyncSession::sync_manager() const
 {
+    std::lock_guard<std::mutex> lk(m_state_mutex);
```

Comment:
Why are we locking the state mutex here? It doesn't appear that `detach_from_sync_manager` is guarded by the same mutex. Or am I missing some more complex interaction?